src/announcements.py

```
@@ -172,6 +173,23 @@ async def get_announcement_channel(
+        # Synchronous path: many command handlers call this from sync context
+        from src.db import get_session
+
+        with get_session() as session:
+            cfg = get_guild_config(session, getattr(guild, "id", None))
+            if cfg and cfg.announcement_channel_id:
+                ch = discord.utils.get(
+                    guild.text_channels, id=cfg.announcement_channel_id
+                )
+                if ch and _can_send(ch, _get_bot_member(guild)):
+                    return ch
+    except Exception:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)